test/match_return.jl

```diff
@@ -0,0 +1,118 @@
+# Macro used in test below
+macro user_match_fail()
+    quote
+        @match_fail
+    end
+end
+
+# Macro used in test below
+macro user_match_return(e)
+    esc(quote
+        @match_return($e)
+    end)
+end
+
+@testset "@match_return tests" begin
+
+@testset "simple uses work correctly" begin
+    @test (@match2 Foo(1, 2) begin
+        Foo(x, 2) => begin
+            x
+            @match_fail
+        end
+        Foo(1, x) => begin
+            @match_return x
+            12
+        end
+    end) == 2
+end
+
+file = Symbol(@__FILE__)
+
+@testset "uses of early-exit macros outside @match2 produce errors 1" begin
+    let line = 0
+        try
+            line = (@__LINE__) + 1
+            @eval @match_return 2
+            @test false
+        catch ex
+            @test ex isa LoadError
+            e = ex.error
+            @test e isa ErrorException
+            @test e.msg == "$file:$line: @match_return may only be used within the value of a @match case."
+        end
+    end
+end
+
+@testset "uses of early-exit macros outside @match2 produce errors 2" begin
+    let line = 0
+        try
+            line = (@__LINE__) + 1
+            @eval @match_fail
```

Comment:
what happens if we pass the incorrect number of arguments to `@match_return` and `@match_fail`?

To be consistent with Julia, I'd expect `@match_return` to be an alias for `@match_return nothing`

Reply:
Adding a test.

Reply:
You would just get a load error, like any function call with the wrong number of arguments.